recipes/gdal/post_3.5.0/patches/3.5.1/0-replace-find-package.patch

```diff
@@ -0,0 +1,287 @@
+-find_package(PROJ 9 CONFIG QUIET)
+-if (NOT PROJ_FOUND)
+-  find_package(PROJ 8 CONFIG QUIET)
+-endif()
++find_package2(PROJ)
++target_include_directories(CONAN_PKG::proj INTERFACE ${PROJ_INCLUDE_DIRS})
++#if (NOT PROJ_FOUND)
++#  find_package(proj 8 CONFIG QUIET)
++#endif()
+ if (PROJ_FOUND)
+   string(APPEND GDAL_IMPORT_DEPENDENCIES "find_dependency(PROJ ${PROJ_VERSION_MAJOR} CONFIG)\n")
+ else()
+-  find_package(PROJ 6.0 REQUIRED)
++  find_package(proj 6.0 REQUIRED)
+   string(APPEND GDAL_IMPORT_DEPENDENCIES "find_dependency(PROJ 6.0)\n")
+ endif ()
+ 
+@@ -379,15 +337,10 @@ gdal_check_package(JSONC "json-c library (external)" CAN_DISABLE
+   TARGETS json-c::json-c JSONC::JSONC
+ )
+ gdal_internal_library(JSONC REQUIRED)
+-if(TARGET json-c::json-c)
+-  get_target_property(include_dirs json-c::json-c INTERFACE_INCLUDE_DIRECTORIES)
+-  find_path(GDAL_JSON_INCLUDE_DIR NAMES json.h PATHS ${include_dirs} PATH_SUFFIXES json-c NO_DEFAULT_PATH)
+-  list(APPEND include_dirs "${GDAL_JSON_INCLUDE_DIR}")
+-  list(REMOVE_DUPLICATES include_dirs)
+-  set_target_properties(json-c::json-c PROPERTIES
+-    INTERFACE_INCLUDE_DIRECTORIES "${GDAL_JSON_INCLUDE_DIR}"
+-  )
+-endif()
++get_target_property(include_dirs CONAN_PKG::json-c INTERFACE_INCLUDE_DIRECTORIES)
++list(APPEND include_dirs "${JSONC_INCLUDE_DIRS}/json-c")
++set_target_properties(CONAN_PKG::json-c PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${include_dirs}")
++message("Setting include for json-c: ${include_dirs}")
+ 
+ gdal_check_package(OpenCAD "libopencad (external, used by OpenCAD driver)" CAN_DISABLE)
+ gdal_internal_library(OPENCAD)
+@@ -482,7 +435,7 @@ if (GDAL_USE_RASTERLITE2)
+ endif ()
+ cmake_dependent_option(GDAL_USE_RASTERLITE2 "Set ON to use Rasterlite2" ON HAVE_RASTERLITE2 OFF)
+ 
+-find_package(LibKML COMPONENTS DOM ENGINE)
++find_package(LibKML COMPONENTS kmlengine kmldom kmlbase)
+ if (GDAL_USE_LIBKML)
+   if (NOT LibKML_FOUND)
+     message(FATAL_ERROR "Configured to use GDAL_USE_LIBKML, but not found")
+diff --git a/cmake/helpers/ConanFindPackage.cmake b/cmake/helpers/ConanFindPackage.cmake
+new file mode 100644
+index 0000000000..4fe1c7aee8
+--- /dev/null
++++ b/cmake/helpers/ConanFindPackage.cmake
+@@ -0,0 +1,40 @@
++
++function(define_find_package2 pkgname include_file library_name)
++endfunction()
++
++function(find_package2 pkgname)
++    set(_options QUIET REQUIRED)
++    set(_oneValueArgs OUT_DEPENDENCY)
++    set(_multiValueArgs)
++    cmake_parse_arguments(arg "${_options}" "${_oneValueArgs}" "${_multiValueArgs}" ${ARGN})
++    if(arg_QUIET)
++        set(${pkgname}_FIND_QUIETLY TRUE)
++    endif()
++    if(arg_REQUIRED)
++        set(${pkgname}_FIND_REQUIRED TRUE)
++    endif()
++
++    string(TOUPPER ${pkgname} key)
++
++    set(docstring "Configured for conan package ${GDAL_CONAN_PACKAGE_FOR_${key}}")
++    if (DEFINED GDAL_CONAN_PACKAGE_FOR_${key})
++        message("Using conan package ${GDAL_CONAN_PACKAGE_FOR_${key}} for dependency ${pkgname}")
```

Comment:
Large C++ projects have the most 🤢build scripts for package managers...

I do not like this but its not deathly painful to understand and follow....

Reply:
removed, it is a left-over.